test/e2e/specs/editor/various/block-bindings.spec.js

```diff
@@ -229,133 +50,230 @@ test.describe( 'Block bindings', () => {
 			test( 'Should show the value of the custom field', async ( {
```

Comment:
Nit: A lot of these tests seem duplicated. We can simply use a for loop to cycle through each block (`paragraph`, `heading`, `image`, etc) to accomplish the same thing with less code. However, I think maybe we only need to test one block since all of them share the same code. The others can be tested in a much faster way like unit tests or integration tests. This also means we don't have to add more tests as we support more blocks in the future. These tests increase the coverage but not necessarily increase the confidence in our code, IMHO.

I don't think it matters though 😅, and we can always refine our tests. It's just an open discussion that we can circle back in the future.

Reply:
I agree that we could possibly reduce the tests. For this initial version, I just wanted to ensure that the initial set of blocks work as expected. Anyway, I believe it's worth considering it for a follow-up PR, maybe when we add support for more blocks?

Reply:
As long as the handling is hardcoded, it's better to have everything covered with detailed e2e tests. In particular, it's important for the frontend where we still can't fully leverage HTML API to replace attributes in the saved HTML. At some point, we will be able to follow what @kevin940726 recommended.